src/main/java/org/folio/bulkops/service/SrsService.java

```diff
@@ -0,0 +1,96 @@
+package org.folio.bulkops.service;
+
+import static org.folio.bulkops.domain.dto.OperationStatusType.COMPLETED;
+import static org.folio.bulkops.domain.dto.OperationStatusType.COMPLETED_WITH_ERRORS;
+import static org.folio.bulkops.domain.dto.OperationStatusType.RETRIEVING_FROM_SRS;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.bulkops.client.RemoteFileSystemClient;
+import org.folio.bulkops.client.SrsClient;
+import org.folio.bulkops.domain.bean.GetParsedRecordsBatchConditions;
+import org.folio.bulkops.domain.bean.GetParsedRecordsBatchRequestBody;
+import org.folio.bulkops.domain.entity.BulkOperation;
+import org.folio.bulkops.repository.BulkOperationRepository;
+import org.marc4j.MarcJsonReader;
+import org.marc4j.marc.Record;
+import org.springframework.stereotype.Service;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class SrsService {
+  private static final int SRS_CHUNK_SIZE = 100;
+
+  private final SrsClient srsClient;
+  private final BulkOperationRepository bulkOperationRepository;
+  private final RemoteFileSystemClient remoteFileSystemClient;
+  private final ErrorService errorService;
+
+  public void retrieveMarcInstancesFromSrs(List<String> instanceIds, BulkOperation bulkOperation) {
+    var path = bulkOperation.getLinkToCommittedRecordsMarcFile();
+    remoteFileSystemClient.remove(path);
+    if (!instanceIds.isEmpty()) {
+      bulkOperation.setStatus(RETRIEVING_FROM_SRS);
+      bulkOperationRepository.save(bulkOperation);
+      var offset = 0;
+      try (var writer = remoteFileSystemClient.marcWriter(path)) {
+        while (offset < instanceIds.size()) {
+          var ids = instanceIds.stream()
+            .skip(offset)
+            .limit(SRS_CHUNK_SIZE)
+            .toList();
+          var marcJsons = srsClient.getParsedRecordsInBatch(new GetParsedRecordsBatchRequestBody(
+            new GetParsedRecordsBatchConditions(ids, "INSTANCE"), "MARC_BIB"))
+            .get("records");
+          for (var jsonNodeIterator = marcJsons.elements(); jsonNodeIterator.hasNext();) {
+            var srsRec = jsonNodeIterator.next();
+            var content = srsRec.get("parsedRecord").get("content").toString();
+            writer.writeRecord(jsonToMarcRecord(content));
+          }
+          offset += ids.size();
+          updateBulkOperationProgress(bulkOperation, offset);
```

Comment:
I don't think it's valid logic, if we had progress from data import and after that you put 0, it will reset progress bar and additional questions will be raised